Assets/Scripts/PlayerScript.cs

```diff
@@ -8,88 +7,133 @@ public class PlayerScript : MonoBehaviour {
+    [SerializeField] protected float multiInputWindow = 0.05f;
+    [SerializeField] protected float dashSpeed;
+    [SerializeField] protected float dashTiming = 0.4f;
+    
+    private float currentSpeed;
+
+    private bool isHorizontalAxisInUse = false;
+    private bool isVerticalAxisInUse = false;
+
+    private float lastInitialDirectionalInputTime;
+    private bool playerInAction = false;
+    private Vector3 currMoveDir;
+    private bool hasDashed = false;
 
     void Awake() {
         if (Instance == null) {
             Instance = this;
         }
     }
 
-    int frameCnt;
-
-    int diagonalTolerance = 5; //in frames
-
-    bool movingHori,movingVert;
-
-    bool movementlockedIn;
-
     void Start() {
         destination.parent = null;
-        movingHori = false;
-        movingVert = false;
-        movementlockedIn = false;
-        targetPos = destination.position;
-        frameCnt = diagonalTolerance;
+        currentSpeed = playerSpeed;
     }
 
-    Vector3 targetPos;
-    public bool wallAtPos(Vector3 pos) {
-        return Physics2D.OverlapCircle(pos, .1f, collisionMask);
-    }
     void Update() {
-        // Move Player to destination point
-        transform.position = Vector3.MoveTowards(transform.position, destination.position, playerSpeed * Time.deltaTime);
-
-        if ( Vector3.Distance(transform.position, destination.position) <= .05f || frameCnt<diagonalTolerance) {   // Check if Player is near destination or 
-            if (Mathf.Abs(Input.GetAxisRaw("Horizontal")) == 1f) {
-                if (!wallAtPos(targetPos + new Vector3(Input.GetAxisRaw("Horizontal"),0,0)) && !movingHori) { // Check if new destination position will overlap with collisionMask
-                    Vector3 horiDir =  new Vector3(Input.GetAxisRaw("Horizontal"), 0f, 0f);
-                    movingHori = true;
-                    if (movingVert) {
-                        targetPos += horiDir;
-                        movementlockedIn = true;
-                    } else {
-                        targetPos = destination.position + horiDir;
+        // Move Player to destination point after input window closes
+        if (Time.time - lastInitialDirectionalInputTime >= multiInputWindow) {
+            transform.position = Vector3.MoveTowards(transform.position, destination.position, currentSpeed * Time.deltaTime);
+        }
+
+        Vector3 currInputDir = new(Input.GetAxisRaw("Horizontal"), Input.GetAxisRaw("Vertical"), 0f);
+
+        // isHorizontalAxisInUse and isVerticalAxisInUse make GetAxisRaw behave like GetKeyDown instead of GetKey
+
+        // Check if an input for a diagonal move was made
+        // This is to add leniency to making a diagonal move so it doesn't have to be frame-perfect
+        if (Time.time - lastInitialDirectionalInputTime < multiInputWindow) {
+            if ((Mathf.Abs(currInputDir.x) == 1f && !isHorizontalAxisInUse) || (Mathf.Abs(currInputDir.y) == 1f && !isVerticalAxisInUse)) {
```

Comment:
style: this conditional is kinda long and is reused a few times, might be good to split this into a separate private function to make it a bit more readable